docs/bigtable-data-api.rst

```diff
@@ -0,0 +1,328 @@
+
+  .. code:: python
+
+      row.delete_cells(column_family_id, [column1, column2],
+                       time_range=time_range)
+
+  In addition, if we want to delete cells from every column in a column family,
+  the special :attr:`ALL_COLUMNS <gcloud.bigtable.row.Row.ALL_COLUMNS>` value
+  can be used
+
+  .. code:: python
+
+      row.delete_cells(column_family_id, Row.ALL_COLUMNS,
+                       time_range=time_range)
+
+* :meth:`delete() <gcloud.bigtable.row.Row.delete>` will delete the entire row
+
+  .. code:: python
+
+      row.delete()
+
+Conditional Mutations
+---------------------
+
+Making **conditional** conditional modifications is essentially identical
+to **direct** modifications, but we need to specify a filter to match
+against in the row:
+
+.. code:: python
+
+    row = table.row(row_key, filter_=filter_val)
+
+See the :class:`Row <gcloud.bigtable.row.Row>` class for more information
+about acceptable values for ``filter_``.
+
+The only other difference from **direct** modifications are that each mutation
+added must specify a ``state``: will the mutation be applied if the filter
+matches or if it fails to match.
+
+For example
+
+.. code:: python
+
+    row.set_cell(column_family_id, column, value,
+                 timestamp=timestamp, state=True)
+
+.. note::
+
+    If ``state`` is passed when no ``filter_`` is set on a
+    :class:`Row <gcloud.bigtable.row.Row>`, adding the mutation will fail.
+    Similarly, if no ``state`` is passed when a ``filter_`` has been set,
+    adding the mutation will fail.
+
+Append Mutations
+----------------
+
+Append mutations can be added via one of two methods
+
+* :meth:`append_cell_value <gcloud.bigtable.row.Row.append_cell_value>` appends
+  a bytes value to an existing cell:
+
+  .. code:: python
+
+      row.append_cell_value(column_family_id, column, bytes_value)
+
+* :meth:`increment_cell_value <gcloud.bigtable.row.Row.increment_cell_value>` increments
+  an integer value in an existing cell:
+
+  .. code:: python
+
+      row.increment_cell_value(column_family_id, column, int_value)
+
+  Since only bytes are stored in a cell, the cell value is decoded as
+  an unsigned 64-bit integer before being incremented. (This happens on
+  the Google Cloud Bigtable server, not in the library.)
+
+Notice that no timestamp was specified. This is because **append** mutations
+operate on the latest value of the specified column.
+
+If there are no cells in the specified column, then the empty string (bytes
+case) or zero (integer case) are the assumed values.
+
+Starting Fresh
+--------------
+
+If accumulated mutations need to be dropped, use
+:meth:`clear_mutations() <gcloud.bigtable.row.Row.clear_mutations>`
+
+.. code:: python
+
+    row.clear_mutations()
+
+To clear **append** mutations, use
+:meth:`clear_modification_rules() <gcloud.bigtable.row.Row.clear_modification_rules>`
+
+.. code:: python
+
+    row.clear_modification_rules()
```

Comment:
Name is based on the name of the protobuf [message type][1]: `ReadModifyWriteRule`. If `commit()` and `commit_modifications()` get combined, then this method would be nuked.

[1]: https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/1ff247c2e3b7cd0a2dd49071b2d95beaf6563092/bigtable-protos/src/main/proto/google/bigtable/v1/bigtable_data.proto#L491